includes/src/Plugin.php

```diff
@@ -0,0 +1,334 @@
+
+                if ($hook === $this->screen) {
+                    wp_enqueue_style($this->plugin_page.'-core', $plugin_url.'includes/admin/exnano.css', null, $version);
+                }
+            }
+        );
+
+        add_action('network_admin_menu', function () {
+            add_submenu_page(
+                $this->menu_parent, // Parent element
+                'Cloudflare DNS Manager', // Text in browser title bar
+                'CF DNS Manager', // Text to be displayed in the menu.
+                'manage_network_options', // Capability
+                $this->plugin_page, // Page slug, will be displayed in URL
+                [$this, 'settings_page'], // Callback function which displays the page
+            );
+        });
+
+        add_filter(
+            'network_admin_plugin_action_links_'.$this->hook,
+            function ($links) {
+                $new = [
+                    'exnwpmucfdnsmanager-settings' => sprintf('<a href="%s">%s</a>', network_admin_url($this->menu_parent.'/?page='.$this->plugin_page), __('Settings', 'exn-wpmu-cf-dns-manager')),
+                ];
+
+                return array_merge($new, $links);
+            }
+        );
+    }
+
+    public function settings_page()
+    {
+        include_once $this->path.'/includes/admin/settings.php';
+    }
+
+    public function settings_save()
+    {
+        check_admin_referer($this->nonce_key);
+
+        update_site_option('exn_cf_api_token', sanitize_text_field($_POST['exn_cf_api_token']));
+
+        $redirect = add_query_arg(
+            [
+            'page' => $this->plugin_page,
+            'updated' => true,
+            ],
+            network_admin_url($this->menu_parent)
+        );
+
+        wp_redirect($redirect);
+
+        exit;
+    }
+
+    /**
+     * gcustom_notices.
+     */
+    public function custom_notices()
+    {
+        if (isset($_GET['page']) && $_GET['page'] == $this->plugin_page && isset($_GET['updated'])) {
+            echo '<div id="message" class="updated notice notice-success is-dismissible"><p>'.esc_html__('Settings updated', 'exn-wpmu-cf-dns-manager').'</p></div>';
+        }
+    }
+
+    /**
+     * get_zones_list.
+     */
+    public function get_zones_list()
+    {
+        $network = get_network();
+        $domain = $network->domain;
+
+        if (\defined('EXNANO_MUCFDNSM_TEST_DOMAIN')) {
+            $domain = EXNANO_MUCFDNSM_TEST_DOMAIN;
+        }
+
+        $response = null;
+        $token = get_site_option('exn_cf_api_token');
+        $url = $this->cf_api_url.'zones/';
+        $args = [
+            'headers' => [
+                'Authorization' => 'Bearer '.$token,
+            ],
+        ];
+
+        if (empty($token)) {
+            return false;
+        }
+
+        // Make request to Cloudflare API zones endpoint.
+        $response = wp_remote_get($url, $args);
+        $response_code = wp_remote_retrieve_response_code($response);
+
+        if (200 === $response_code) {
```

Comment:
This should be the line where we must check for `is_wp_error()`, right?

Reply:
need to split this code into another class.. and then will handle it properly.. will submit another prr later

Reply:
ahhh. ok2.